### PR TITLE
Added properties pane to Resource Viewer

### LINKED ISSUE #2409: Resource Viewer: show resource details in details pane
We should show resource details for currently selected resource inside the details pane on the right. Something like this: 

Screen Shot 2021-05-03 at 10 24 07 AM

NOTE: This has been implemented on earlier [experimental branch](https://github.com/mklanjsek/octant/tree/resources-for-realz), but in needs to be brought to current code base and expand on fields we display for each resource.

### PR DESCRIPTION
**Pull request by mklanjsek**
Properties are now displayed for selected node in Resource Viewer: 

Screen Shot 2021-05-20 at 2 20 39 PM

Properties are created and merged together like this:
1. `Version` and `Kind` are always present at the top. These were already passed down to client, so they are simply just directly added there,
2. `Namespace`, `Created` and `Labels` (if present) are added for all resource types,
3. Next group depends on resource type, for example for Pods we are adding `Service Account`, `Node` and `Controlled By` fields,
4. Lastly, at the very bottom are properties that are added by plugins. In screenshot above, you can see that our sample plugin has added the `ID` field (example how to do that is included for plugin authors),

**Special notes for your reviewer**:
Please let me know if there are certain properties that are better suited for a specific resource. We may need a second pass to tweak what exactly shows up for each resource type.

Signed-off-by: Milan Klanjsek <mklanjsek@pivotal.io>

- Fixes #2409

### CODE REVIEW

**Review by xtreme-vikram-yadav — CHANGES_REQUESTED** [review]
This is an amazing new feature. Thank you for bringing it in.

File: internal/objectstatus/deployment.go [hunk 1]
```diff
@@ -51,6 +57,28 @@ func deploymentAppsV1(_ context.Context, object runtime.Object, _ store.Store) (
 				component.NewText(
 					fmt.Sprintf("Expected %d replicas, but %d are available",
 						status.Replicas, status.AvailableReplicas))},
+			Properties: properties,
 		}, nil
 	}
 }
+
+func getSelector(deployment *appsv1.Deployment) component.Component {
```
[the remark this answers is not in the document]
Comment (mklanjsek) [reply internal/objectstatus/deployment.go]: Great point, thanks. Refactored to make it behave same way.

File: internal/objectstatus/pod.go [hunk 2]
```diff
@@ -48,5 +52,29 @@ func pod(ctx context.Context, object runtime.Object, o store.Store) (ObjectStatu
...
+		if ownerReference != nil {
+			controlledBy, err := link.ForGVK(
+				pod.Namespace,
+				ownerReference.APIVersion,
+				ownerReference.Kind,
+				ownerReference.Name,
+				ownerReference.Name,
+			)
+			if err == nil {
+				status.Properties = append(status.Properties, component.Property{Label: "Controlled By", Value: controlledBy})
```
[the remark this answers is not in the document]
Comment (mklanjsek) [reply internal/objectstatus/pod.go]: ✅

File: internal/objectstatus/pod.go [hunk 3]
```diff
@@ -48,5 +52,29 @@ func pod(ctx context.Context, object runtime.Object, o store.Store) (ObjectStatu
 		status.Details = append(status.Details, component.NewText("Ephemeral container is running"))
 	}
 
+	if link != nil {
+		serviceAccountLink, _ := link.ForGVK(pod.Namespace, "v1", "ServiceAccount", pod.Spec.ServiceAccountName, pod.Spec.ServiceAccountName)
+		status.Properties = []component.Property{{Label: "ServiceAccount", Value: serviceAccountLink}}
+
+		if nodeName := pod.Spec.NodeName; nodeName != "" {
+			nodeLink, _ := link.ForGVK("", "v1", "Node", pod.Spec.NodeName, pod.Spec.NodeName)
+			status.Properties = append(status.Properties, component.Property{Label: "Node", Value: nodeLink})
```
[the remark this answers is not in the document]
Comment (mklanjsek) [reply internal/objectstatus/pod.go]: ✅

**Comment by mklanjsek** [comment]
@xtreme-vikram-yadav PTAL

**Review by xtreme-vikram-yadav — APPROVED** [review]

**Comment by xtreme-vikram-yadav** [comment]
Please squash the commits before merging.

**Pull request merged**